viam-orb-slam3.go

```diff
@@ -489,11 +489,25 @@ func (orbSvc *orbslamService) GetSLAMProcessConfig() pexec.ProcessConfig {
 	args = append(args, "-port="+orbSvc.port)
 	args = append(args, "--aix-auto-update")
 
+	target := orbSvc.executableName
+
+	appDir := os.Getenv("APPDIR")
+
+	if appDir != "" {
+		// The orb grpc server is expected to be in
+		// /usr/bin/ if we are running in an appimage.
+		target = appDir + "/usr/bin/" + strings.TrimPrefix(target, "/")
+	}
+
 	return pexec.ProcessConfig{
-		ID:      "slam_orbslamv3",
-		Name:    orbSvc.executableName,
-		Args:    args,
-		Log:     true,
+		ID:   "slam_orbslamv3",
+		Name: target,
+		Args: args,
+		Log:  true,
+		// In appimage this is set to the appimage
+		// squashfs mount location (/tmp/.mountXXXXX)
+		// Otherwise, it is an empty string
+		CWD:     os.Getenv("APPRUN_RUNTIME"),
```

Comment:
Same here

Reply:
@zaporter-work found that this line was required in order to get the app image to be able to boot another executable within the same app image.

It is equal to `/tmp/.mount_orb-slfcenhr/runtime/compat`

I have not been able to find any documentation on the app image docs describing this env var. 

I'll post the output of the error one gets when it is left out.

Reply:
As with other comment above, this is the other part of the "magic" that appimage normally inserts into libc stuff, but doesn't work with Go. See my AIX wrapper (which provides self-update and install functions within our appimages) for another example of doing this. 

https://github.com/Otterverse/aix/blob/main/aix.go#L28-L37

Reply:
So for anyone confused by that error, what's happening is not that the file ITSELF isn't being found, it's that when trying to execute the file, the interpreter (normally `/lib/ld-linux-x86_64.so.2`) isn't being found, as appimage replaced it with a RELATIVE path, so it's looking for `lib/ld-linux-x86_64.so.2` (or the aarch64 version) and thus only runs when in the correct directory. This was a hack that allows appimage to switch runtimes without modifying files, and just changing directories.

Reply:
Thanks very much @Otterverse ! Summarized my understanding for this and the thread above here: https://github.com/viamrobotics/viam-orb-slam3/pull/38/files#r1169049999.